tests/stockfish/test_models.py

```diff
@@ -270,3 +270,33 @@ def test_get_parameters(self):
         arg1 = s1.get_parameters()
         arg2 = s2.get_parameters()
         assert arg1 != arg2
+
+    def test_get_top_moves(self):
+        stockfish = Stockfish(depth = 15, parameters = {"MultiPV": 4})
+        stockfish.set_fen_position("1rQ1r1k1/5ppp/8/8/1R6/8/2r2PPP/4R1K1 w - - 0 1")
+        assert stockfish.get_top_moves(2) == [{"Move": "e1e8", "Centipawn": None, "Mate": 1},
+                                              {"Move": "c8e8", "Centipawn": None, "Mate": 2}]
+        stockfish.set_fen_position("8/8/8/8/8/3r2k1/8/6K1 w - - 0 1")
+        assert stockfish.get_top_moves(2) == [{"Move": "g1f1", "Centipawn": None, "Mate": -2},
+                                              {"Move": "g1h1", "Centipawn": None, "Mate": -1}]
+    
+    def test_get_top_moves_mate(self):
+        stockfish = Stockfish(depth = 10, parameters = {"MultiPV": 3})
+        stockfish.set_fen_position("8/8/8/8/8/6k1/8/3r2K1 w - - 0 1")
+        assert stockfish.get_top_moves() == []
+        assert stockfish.get_parameters()["MultiPV"] == 3
+    
+    def test_get_top_moves_raising_error(self):
+        stockfish = Stockfish()
+        stockfish.set_fen_position("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1")
+        raised_error = False
+        try:
```

Comment:
It should be better to check that exception was raised with this construction https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions

```
import pytest

def test_zero_division():
    with pytest.raises(ZeroDivisionError):
        1 / 0
```